src/main/java/frc/robot/subsystems/ArmRotate.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.subsystems;
+
+import java.util.function.DoubleSupplier;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.chopshop166.chopshoplib.commands.SmartSubsystemBase;
+import com.chopshop166.chopshoplib.motors.CSSparkMax;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
+import com.chopshop166.chopshoplib.sensors.IEncoder;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.maps.subsystems.ArmRotateMap;
+import frc.robot.maps.subsystems.ArmRotateMap.Data;
+
+public class ArmRotate extends SmartSubsystemBase {
+
+    private SmartMotorController armMotor = new CSSparkMax(1, MotorType.kBrushless);
+    private IEncoder armAngle;
+    private ArmRotateMap map;
+    private PIDController pid = new PIDController(0, 0, 0);
+    final double MOVE_SPEED = 0.5;
+    final double COMPARE_ANGLE = 5;
+    final double TOP_ANGLE = 85;
+    final double BOTTOM_ANGLE = 5;
+    final double SLOW_DOWN = 0.1;
+    final double NO_POWER = 0;
+
+    Data data;
+
+    public ArmRotate(ArmRotateMap map) {
+
+        this.map = map;
+    }
+
+    public CommandBase Move(DoubleSupplier rotationSpeed) {
+        return run(() -> {
+            data.setPoint = rotationSpeed.getAsDouble();
+        });
+    }
+
+    public CommandBase moveToAngle(double angle) {
+        // When executed the arm will move. The encoder will update until the desired
+        // value is reached, then the command will end.
+        return cmd("Move To Set Angle").onExecute(() -> {
+            data.setPoint = pid.calculate(armAngle.getDistance() - angle);
```

Comment:
- Is `armAngle.getDistance` in the same unit as `angle`?
- If you make the lowest possible angle `0.0`, then any rotation is relative to that point. You can use this to avoid doing some suspicious math.